presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSink.java

```diff
@@ -141,18 +141,16 @@ public RaptorPageSink(
     @Override
     public CompletableFuture<Collection<Slice>> finish()
     {
-        List<ShardInfo> shards = new ArrayList<>();
-        for (PageBuffer pageBuffer : pageWriter.getPageBuffers()) {
+        List<CompletableFuture<? extends List<Slice>>> futureSlices = pageWriter.getPageBuffers().stream().map(pageBuffer -> {
```

Comment:
If I don't use "extends" then it complains about unchecked cast. I think it's because of how "allAsList()" method is defined.